test/hotspot/jtreg/compiler/integerArithmetic/TestOrSaturate.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+ package compiler.integerArithmetic;
+
+ import compiler.lib.ir_framework.*;
+ import jdk.test.lib.Asserts;
+
+ /*
+  * @test
+  * @bug 8352839
+  * @summary Test that an or with all bits set is folded to all bits (x | -1 == -1).
+  * @library / /test/lib
+  * @run driver compiler.integerArithmetic.TestOrSaturate
+  */
+
+public class TestOrSaturate {
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    private final static int WARMUP = 10_000;
+
+    @Run(test = {"testL", "testI", "testDelayed"})
+    public static void check() {
+        for (int i = 0; i < WARMUP; i++) {
```

Comment:
Do you actually need the WARMUP here? It does not look like this is a standalone test, so I thought the `@Run` actually gets called many times. Not sure if that is correct...

Reply:
The name `WARMUP` is not strictly correct, but I left the loop to test more than one value.

Reply:
Changed to random value in [e02adc4](https://github.com/openjdk/jdk/pull/24289/commits/e02adc43ccfa4c944f646b93d311115cbce2589a)